rts/motoko-rts/src/custom.rs

```diff
@@ -0,0 +1,326 @@
+// Custom RTS function utilities
+
+use alloc::vec::Vec;
+use motoko_rts_macros::{motoko, tuple_macro};
+
+use crate::{
+    barriers::allocation_barrier,
+    memory::{alloc_array, alloc_blob, Memory},
+    types::{Bytes, Tag, Value, TAG_ARRAY, TAG_BLOB},
+};
+
+extern "C" {
+    fn unit() -> Value;
+
+    fn u8_from_nat8(value: Value) -> u8;
+    fn u16_from_nat16(value: Value) -> u16;
+    fn u32_from_nat32(value: Value) -> u32;
+    fn u64_from_nat64(value: Value) -> u64;
+
```

Comment:
Maybe, we could also add `float` conversions?

Reply:
'char' would be useful too.